client/pages/plant-detail.jsx

```diff
@@ -1,25 +1,29 @@
 import React from 'react';
 import GardenForm from '../components/garden-form';
+import DeleteConfirmation from '../components/delete-confirmation';
 
 export default class PlantDetail extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       plant: null,
-      btnText: 'Add to garden',
+      btnText: null,
       gardenCreated: null,
       gardenId: null,
-      modalClass: 'hidden',
+      gardenFormClass: 'hidden',
+      deleteConfirmationClass: 'hidden',
```

Comment:
Instead of storing CSS classes and text in state, define separate methods that `return` the desired values and call them on-demand in your `render` method.

```js
getButtonText() {

}

getGardenFormClass() {
 
}

getDeleteConfirmationClass() {

}
```

Try to only store facts in your state, like:
- is the modal open?
- details about the plant
- is the plant in the garden (which i believe the plant object contains)

This will simplify your code and make it easier to follow.